### ISSUE TITLE
Dependencies are overly constraining

### ISSUE THREAD

**Issue by kevin218** [spacetelescope/pastasoss#13]
The environment.yml files specify unnecessarily specific versions for all packages.  For example, PASTASOSS supposedly requires `numpy=1.23.4` and `scipy=1.10.1`.  Using `=` instead of `>` or `<` in the dependencies doesn't future-proof the code and makes it incredibly difficult to integrate PASTASOSS into other pipelines.  I encourage the developers to **only** apply constraints to version numbers when absolutely necessary.  Otherwise, just use `numpy` and `scipy` in the list of dependencies.

Touching specifically on my issue, PyMC3 requires `numpy[version='>=1.20.0,<1.22']`.  So, at minimum, relaxing the version constraint for numpy would be helpful.

**Comment by kevin218** [comment]
It looks like #14 resolves this issue.  Can we get it merged into main soon so that I can test the changes in Eureka!?  Thanks.

**Comment by tbainesUA** [comment]
Hi Kevin, 

We've made the changes so you can now pull down the changes. 

- Tyler

**Comment by kevin218** [comment]
Hi Tyler,

I'm still having issues with the scipy version dependency:
``jwst 1.11.4 depends on scipy<1.10.0 and >=1.6.0``
``pastasoss 1.2.1 depends on scipy>=1.10.1``
It looks like the only scipy function you're using is interp1d, which requires ``scipy>=1.7.1``.

**Comment by tbainesUA** [comment]
Hi Kevin, 

Is there a particular need for retaining build 1.11.4 for your code? 

The current build of the `jwst` is `v1.17.4` while yours is showing build `v1.11.4` which is out of date. Updating `jwst` to the latest build should resolve the issue. We've tested this ourselves in a blank environment and there's no conflict. 

-Tyler

**Comment by kevin218** [comment]
Yes and no. We have two different versions of Eureka! that can be installed.  The one that uses PyMC3 has some annoying requirements that prevent us from using newer versions of the `jwst` pipeline.  The other version works just fine with the current build.

**Comment by tbainesUA** [comment]
Hello Kevin,

Apologies in delay. 

We have updated the package requirements and adjusted dependency versions. Scipy has been downgraded to scipy>=1.8.0, which is the minimum version required to retain compatibility with Python 3.10 and adhere to ST code standards while also meeting your requirements.

A new version of pastasoss 1.2.2 was released to reflect these changes. 

Please note that we cannot guarantee this long-term, as we may be required to update the numpy version restrictions in accordance with ST standards and to properly support future versions of the JWST pipeline. 

If you have any addition questions feel free to ask. 

Tyler

**Comment by kevin218** [comment]
Hi Tyler,

Thank you for implementing the change. I can confirm that all of our tests now pass.

Also, I understand if the requirements change down the road.
